### PR TITLE
ref(grouping): Refactor similarity utils

### PR DESCRIPTION
**Pull request by armenzg**
It makes reading the code a little easier.

### CODE REVIEW

File: src/sentry/seer/similarity/utils.py [hunk 1]
```diff
@@ -197,59 +205,15 @@ def get_stacktrace_string(data: dict[str, Any], platform: str | None = None) ->
...
-    frame_count = 0
-    html_frame_count = 0  # for a temporary metric
-    is_frames_truncated = False
-    has_no_filename_or_module = False
-    stacktrace_str = ""
-    found_non_snipped_context_line = False
-
     metrics.distribution("seer.grouping.exceptions.length", len(exceptions))
 
-    def _process_frames(frames: list[dict[str, Any]]) -> list[str]:
```
Comment (armenzg) [inline src/sentry/seer/similarity/utils.py:209]: Now that we use a dictionary to track the metrics it is not as ugly to take this function out of here.

File: src/sentry/seer/similarity/utils.py [hunk 2]
```diff
@@ -266,31 +230,52 @@ def _process_frames(frames: list[dict[str, Any]]) -> list[str]:
...
-        if has_no_filename_or_module:
+        if processing_metrics["has_no_filename_or_module"]:
             raise NoFilenameOrModuleException
         # Only exceptions have the type and value properties, so we don't need to handle the threads
         # case here
         header = f"{exc_type}: {exc_value}\n" if exception["id"] == "exception" else ""
 
         result_parts.append((header, frame_strings))
 
+    return generate_stacktrace_string(
```
Comment (armenzg) [inline src/sentry/seer/similarity/utils.py:266]: All of this code has been moved within the function:
image

**Review by jangjodi — APPROVED** [review]
this looks great! just letting you know that I added another frame metric [here](https://github.com/getsentry/sentry/pull/81700/files) that will need to be added to the `FrameMetrics` dictionary

**Comment by armenzg** [comment]
> this looks great! just letting you know that I added another frame metric [here](https://github.com/getsentry/sentry/pull/81700/files) that will need to be added to the `FrameMetrics` dictionary

Thanks for letting me know! I added it.

**Pull request merged**